src/helpers/configuration_helpers.sh

```diff
@@ -21,6 +21,9 @@ function add_non_feature_branch {
   elif [ "$(is_non_feature_branch "$branch_name")" = true ]; then
     echo_inline_error "'$branch_name' is already a non-feature branch"
     exit 1
+  elif [ "$branch_name" == "$main_branch_name" ]; then
+    echo_inline_error "'$branch_name' is already set as the main branch"
+    exit 1
```

Comment:
Part of this is exit_with_error adds a new line (a stupid implementation of exit_with_error by me). I will rename that function so exit_with_error only exits and we have another function for adding a newline and exiting with error